packages/catalog-client/src/types.ts

```diff
@@ -17,7 +17,10 @@
 import { Entity, EntityName, Location } from '@backstage/catalog-model';
 
 export type CatalogEntitiesRequest = {
-  filter?: Record<string, string | string[]> | undefined;
+  filter?:
+    | Record<string, string | string[]>[]
+    | Record<string, string | string[]> // Legacy type preserved for backwards compatibility
```

Comment:
Could skip this comment tbh, it's there for convenience to not have to always wrap in an array